nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -792,7 +792,10 @@ in
         fi
       '';
 
-    nix.nrBuildUsers = mkDefault (max 32 (if cfg.settings.max-jobs == "auto" then 0 else cfg.settings.max-jobs));
+    nix.nrBuildUsers = mkDefault (
+      if cfg.settings.auto-allocate-uids or false then 0
```

Comment:
Not sure I like having options depend on `settings` as those are usually only passed to the application rather than used in NixOS config.

Reply:
Well, the fact that we're able to reference arbitrary settings in the configuration is one of the main [motivations](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#part-1) of doing things this way. Do you have a specific concern in mind?

The `or false` does feel a bit ugly, that could be fixed by making `auto-allocate-uids` an explicit option, but this is an experimental feature so maybe we don't want that.

Reply:
I don't see anything regarding modules depending on certain settings in that section.

From my view, `settings` is supposed to be converted into a string or file in a generic way; handling every attribute the same. It just lends itself better for that purpose than a collection of strings and I fully agree with the RFC on that.  
I've never seen a module depend on one particular setting however which is why I thought it was notable.

Reply:
That's how i interpret
> Values within it can't be inspected, since it's just an opaque string

Note that this option already depends on `cfg.settings.max-jobs`.